### PR TITLE
android scroll performance fixes pt. 1

### PR DESCRIPTION
**Pull request by haileyok**
## Why

Investigating Android performance problems led us to - at least in part - the lack of memoization here. There is a considerable improvement by memoizing here.

## Test Plan

Essentially, checkout main and test from a low end Android device. Compare that to what this branch feels like. There should be a marked difference in performance.

Also worth checking - perhaps side by side with a different device running on main - to make sure that no styles have changed by doing this, though they shouldn't have.

### CODE REVIEW

File: src/alf/index.tsx [hunk 1]
```diff
@@ -103,35 +103,32 @@ export function ThemeProvider({
...
+      themes,
+      fontScale,
+      setFontScaleAndPersist,
+      fontFamily,
+      setFontFamilyAndPersist,
+      fontScaleMultiplier,
+    ],
   )
+
+  return <Context.Provider value={value}>{children}</Context.Provider>
```
Comment (haileyok) [inline src/alf/index.tsx:131]: no change here, just moving this out of the jsx

File: src/view/com/util/text/Text.tsx [hunk 2]
```diff
@@ -74,49 +75,45 @@ export function Text({
     // @ts-ignore
     if (flattened.fontSize) {
       // @ts-ignore
-      flattened.fontSize = flattened.fontSize * fonts.scaleMultiplier
+      flattened.fontSize = Math.round(
```
Comment (haileyok) [inline src/view/com/util/text/Text.tsx:78]: Rounding here might not be strictly necessary, but it feels bad to have a chance of a value like `14.065`, which is one of the font sizes that we observe in the main feed with some font preferences.

File: src/view/com/util/text/Text.tsx [hunk 3]
```diff
@@ -74,49 +75,45 @@ export function Text({
...
+      )
     }
 
-    const shared = {
-      uiTextView: true,
+    return {
+      uiTextView: selectable && isIOS,
       selectable,
       style: flattened,
+      dataSet: Object.assign({tooltip: title}, dataSet || {}),
```
Comment (gaearon) [inline src/view/com/util/text/Text.tsx:88]: this is now always assigned but used to be skipped under iOS && selectable, right? seems fine either way cause it’s probably web only

**Review by gaearon — APPROVED** [review]

File: src/view/com/util/text/Text.tsx (hunk 3 above)
Comment (haileyok) [reply src/view/com/util/text/Text.tsx:88]: Yea, it should definitely be web only. Actually may make sense to _only_ add the prop for `isWeb` anyway.

**Pull request merged**